### ISSUE TITLE
An exception raised after pasting a BibTeX source to an entry in a subgroup

### ISSUE THREAD

**Issue by dsoutw** [JabRef/jabref#8207]
### JabRef version

5.3 (latest release)

### Operating system

Windows

### Details on version and operating system

Windows 10

### Checked with the latest development build

- [X] I made a backup of my libraries before testing the latest development version.
- [X] I have tested the latest development version and the problem persists

### Steps to reproduce the behaviour

1. Add a subgroup "Subgroup"
2. Add an empty new article entry to the subgroup
3. Switch to the "BibTeX source" 
4. Replace the "BibTeX source" by an article that does NOT contain the "groups" field.
5. An exception will raise after clicking an empty row BELOW the new article entry. 

Remark:
1. In step 4, the exception will NOT raise if pasting a "BibTeX source" with a "groups" field 
2. In step 5, the exception will NOT raise if clicking a nonempty entry

### Appendix

![圖片](https://user-images.githubusercontent.com/31974255/139841524-8b582dc7-2393-458d-918c-57b87d45246a.png)
<details>

<summary>Log File</summary>

java.lang.ArrayIndexOutOfBoundsException: arraycopy: length -1 is negative
	at java.base/java.lang.System.arraycopy(Native Method)
	at org.jabref.merged.module@5.3.60000/javafx.collections.transformation.SortedList.updateUnsorted(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.collections.transformation.SortedList.sourceChanged(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.collections.transformation.TransformationList.lambda$getListener$0(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.collections.WeakListChangeListener.onChanged(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.javafx.collections.ListListenerHelper$Generic.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.javafx.collections.ListListenerHelper.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.collections.ObservableListBase.fireChange(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.collections.ListChangeBuilder.commit(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.collections.ListChangeBuilder.endChange(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.collections.ObservableListBase.endChange(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.collections.transformation.FilteredList.sourceChanged(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.collections.transformation.TransformationList.lambda$getListener$0(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.collections.WeakListChangeListener.onChanged(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.javafx.collections.ListListenerHelper$SingleChange.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.javafx.collections.ListListenerHelper.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.collections.ObservableListBase.fireChange(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.collections.ListChangeBuilder.commit(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.collections.ListChangeBuilder.endChange(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.collections.ObservableListBase.endChange(Unknown Source)
	at com.tobiasdiez.easybind@2.2/com.tobiasdiez.easybind.MappedBackedList.sourceChanged(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.collections.transformation.TransformationList.lambda$getListener$0(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.collections.WeakListChangeListener.onChanged(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.javafx.collections.ListListenerHelper$SingleChange.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.javafx.collections.ListListenerHelper.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.collections.ObservableListBase.fireChange(Unknown Source)
	at org.jabref@5.3.60000/org.jabref.gui.util.UiThreadList.sourceChanged(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.collections.transformation.TransformationList.lambda$getListener$0(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.collections.WeakListChangeListener.onChanged(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.javafx.collections.ListListenerHelper$SingleChange.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.javafx.collections.ListListenerHelper.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.collections.ObservableListBase.fireChange(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.collections.FXCollections$UnmodifiableObservableListImpl.lambda$new$0(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.collections.WeakListChangeListener.onChanged(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.javafx.collections.ListListenerHelper$Generic.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.javafx.collections.ListListenerHelper.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.collections.FXCollections$SynchronizedObservableList.lambda$new$0(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.collections.WeakListChangeListener.onChanged(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.javafx.collections.ListListenerHelper$SingleChange.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.javafx.collections.ListListenerHelper.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.collections.ObservableListBase.fireChange(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.collections.ListChangeBuilder.commit(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.collections.ListChangeBuilder.endChange(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.collections.ObservableListBase.endChange(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.javafx.collections.ObservableListWrapper.access$200(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.javafx.collections.ObservableListWrapper$1$1.invalidated(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.javafx.collections.MapListenerHelper$Generic.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.javafx.collections.MapListenerHelper.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.javafx.collections.ObservableMapWrapper.callObservers(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.javafx.collections.ObservableMapWrapper.put(Unknown Source)
	at org.jabref@5.3.60000/org.jabref.model.entry.BibEntry.setField(Unknown Source)
	at org.jabref@5.3.60000/org.jabref.model.entry.BibEntry.setField(Unknown Source)
	at org.jabref@5.3.60000/org.jabref.gui.entryeditor.SourceTab.storeSource(Unknown Source)
	at org.jabref@5.3.60000/org.jabref.gui.entryeditor.SourceTab.bindToEntry(Unknown Source)
	at org.jabref@5.3.60000/org.jabref.gui.entryeditor.EntryEditorTab.notifyAboutFocus(Unknown Source)
	at org.jabref@5.3.60000/org.jabref.gui.entryeditor.EntryEditor.setEntry(Unknown Source)
	at java.base/java.util.Optional.ifPresent(Unknown Source)
	at org.jabref@5.3.60000/org.jabref.gui.LibraryTab.lambda$createMainTable$15(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.javafx.collections.ListListenerHelper$Generic.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.javafx.collections.ListListenerHelper.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.collections.ObservableListBase.fireChange(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.collections.ListChangeBuilder.commit(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.collections.ListChangeBuilder.endChange(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.collections.ObservableListBase.endChange(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.javafx.scene.control.SelectedItemsReadOnlyObservableList.lambda$new$0(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.javafx.collections.ListListenerHelper$Generic.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.javafx.collections.ListListenerHelper.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.collections.ObservableListBase.fireChange(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.collections.ListChangeBuilder.commit(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.collections.ListChangeBuilder.endChange(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.collections.ObservableListBase.endChange(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.javafx.scene.control.ReadOnlyUnbackedObservableList._endChange(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.scene.control.MultipleSelectionModelBase$SelectedIndicesList._endChange(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.scene.control.ControlUtils.updateSelectedIndices(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.scene.control.TableView$TableViewArrayListSelectionModel.fireCustomSelectedCellsListChangeEvent(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.javafx.collections.ListListenerHelper$SingleChange.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.javafx.collections.ListListenerHelper.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.collections.ObservableListBase.fireChange(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.collections.ListChangeBuilder.commit(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.collections.ListChangeBuilder.endChange(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.collections.ObservableListBase.endChange(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.collections.transformation.SortedList.sourceChanged(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.collections.transformation.TransformationList.lambda$getListener$0(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.collections.WeakListChangeListener.onChanged(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.javafx.collections.ListListenerHelper$SingleChange.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.javafx.collections.ListListenerHelper.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.collections.ObservableListBase.fireChange(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.collections.ListChangeBuilder.commit(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.collections.ListChangeBuilder.endChange(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.collections.ObservableListBase.endChange(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.collections.ModifiableObservableListBase.add(Unknown Source)
	at java.base/java.util.AbstractList.add(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.javafx.scene.control.SelectedCellsMap.add(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.scene.control.TableView$TableViewArrayListSelectionModel.select(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.scene.control.TableView$TableViewArrayListSelectionModel.select(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.scene.control.MultipleSelectionModelBase.shiftSelection(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.scene.control.MultipleSelectionModelBase.shiftSelection(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.scene.control.TableView$TableViewArrayListSelectionModel.updateSelection(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.scene.control.TableView$TableViewArrayListSelectionModel.lambda$new$2(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.collections.WeakListChangeListener.onChanged(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.javafx.collections.ListListenerHelper$Generic.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.javafx.collections.ListListenerHelper.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.collections.ObservableListBase.fireChange(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.collections.transformation.SortedList.sourceChanged(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.collections.transformation.TransformationList.lambda$getListener$0(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.collections.WeakListChangeListener.onChanged(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.javafx.collections.ListListenerHelper$Generic.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.javafx.collections.ListListenerHelper.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.collections.ObservableListBase.fireChange(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.collections.ListChangeBuilder.commit(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.collections.ListChangeBuilder.endChange(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.collections.ObservableListBase.endChange(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.collections.transformation.FilteredList.sourceChanged(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.collections.transformation.TransformationList.lambda$getListener$0(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.collections.WeakListChangeListener.onChanged(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.javafx.collections.ListListenerHelper$SingleChange.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.javafx.collections.ListListenerHelper.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.collections.ObservableListBase.fireChange(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.collections.ListChangeBuilder.commit(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.collections.ListChangeBuilder.endChange(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.collections.ObservableListBase.endChange(Unknown Source)
	at com.tobiasdiez.easybind@2.2/com.tobiasdiez.easybind.MappedBackedList.sourceChanged(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.collections.transformation.TransformationList.lambda$getListener$0(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.collections.WeakListChangeListener.onChanged(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.javafx.collections.ListListenerHelper$SingleChange.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.javafx.collections.ListListenerHelper.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.collections.ObservableListBase.fireChange(Unknown Source)
	at org.jabref@5.3.60000/org.jabref.gui.util.UiThreadList.sourceChanged(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.collections.transformation.TransformationList.lambda$getListener$0(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.collections.WeakListChangeListener.onChanged(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.javafx.collections.ListListenerHelper$SingleChange.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.javafx.collections.ListListenerHelper.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.collections.ObservableListBase.fireChange(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.collections.FXCollections$UnmodifiableObservableListImpl.lambda$new$0(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.collections.WeakListChangeListener.onChanged(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.javafx.collections.ListListenerHelper$Generic.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.javafx.collections.ListListenerHelper.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.collections.FXCollections$SynchronizedObservableList.lambda$new$0(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.collections.WeakListChangeListener.onChanged(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.javafx.collections.ListListenerHelper$SingleChange.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.javafx.collections.ListListenerHelper.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.collections.ObservableListBase.fireChange(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.collections.ListChangeBuilder.commit(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.collections.ListChangeBuilder.endChange(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.collections.ObservableListBase.endChange(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.javafx.collections.ObservableListWrapper.access$200(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.javafx.collections.ObservableListWrapper$1$1.invalidated(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.javafx.collections.MapListenerHelper$Generic.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.javafx.collections.MapListenerHelper.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.javafx.collections.ObservableMapWrapper.callObservers(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.javafx.collections.ObservableMapWrapper.remove(Unknown Source)
	at org.jabref@5.3.60000/org.jabref.model.entry.BibEntry.clearField(Unknown Source)
	at org.jabref@5.3.60000/org.jabref.model.entry.BibEntry.clearField(Unknown Source)
	at org.jabref@5.3.60000/org.jabref.gui.entryeditor.SourceTab.storeSource(Unknown Source)
	at org.jabref@5.3.60000/org.jabref.gui.entryeditor.SourceTab.lambda$setupSourceEditor$5(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.javafx.binding.ExpressionHelper$Generic.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.javafx.binding.ExpressionHelper.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.beans.property.ReadOnlyBooleanPropertyBase.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.scene.Node$FocusedProperty.notifyListeners(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.scene.Scene$12.invalidated(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.beans.property.ObjectPropertyBase.markInvalid(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.beans.property.ObjectPropertyBase.set(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.scene.Scene$KeyHandler.setFocusOwner(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.scene.Scene$KeyHandler.requestFocus(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.scene.Scene.requestFocus(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.scene.Node.requestFocus(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.javafx.scene.control.behavior.TableViewBehaviorBase.mousePressed(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.javafx.scene.control.inputmap.InputMap.handle(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.javafx.event.CompositeEventHandler$NormalEventHandlerRecord.handleBubblingEvent(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.javafx.event.CompositeEventDispatcher.dispatchBubblingEvent(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.javafx.event.EventUtil.fireEventImpl(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.javafx.event.EventUtil.fireEvent(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.event.Event.fireEvent(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.scene.Scene$MouseHandler.process(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.scene.Scene.processMouseEvent(Unknown Source)
	at org.jabref.merged.module@5.3.60000/javafx.scene.Scene$ScenePeerListener.mouseEvent(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.javafx.tk.quantum.GlassViewEventHandler$MouseEventNotification.run(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.javafx.tk.quantum.GlassViewEventHandler$MouseEventNotification.run(Unknown Source)
	at java.base/java.security.AccessController.doPrivileged(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.javafx.tk.quantum.GlassViewEventHandler.lambda$handleMouseEvent$2(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.javafx.tk.quantum.QuantumToolkit.runWithoutRenderLock(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.javafx.tk.quantum.GlassViewEventHandler.handleMouseEvent(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.glass.ui.View.handleMouseEvent(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.glass.ui.View.notifyMouse(Unknown Source)
	at org.jabref.merged.module@5.3.60000/com.sun.glass.ui.win.WinApplication._runLoop(Native Method)
	at org.jabref.merged.module@5.3.60000/com.sun.glass.ui.win.WinApplication.lambda$runLoop$3(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)

</details>

**Comment by ThiloteE** [comment]
Issue #8012 could play a role too, but seems to be slightly different. I can't see 
```
at org.jabref.model.entry.BibEntry.setField(BibEntry.java:563) ~[main_/:?]
at org.jabref.model.entry.BibEntry.setField(BibEntry.java:581) ~[main_/:?]
at org.jabref.gui.entryeditor.SourceTab.storeSource(SourceTab.java:307) ~[main_/:?]
at org.jabref.gui.entryeditor.SourceTab.bindToEntry(SourceTab.java:237) ~[main_/:?]
```
which Christoph identified as likely cause in the logs provided here by dsoutw.


Issue #7999 is different in that the exception will ALSO raise if replacing the groupname within "BibTeX source" with a new groupname. There, not the whole "groupfield" is replaced, but only the data within.

**Comment by ThiloteE** [comment]
There is currently a refactor of the groupsidepane in the making: https://github.com/JabRef/jabref/pull/8202
@HoussemNasri do you think you are touching code related to this issue? Would it be advised to wait for your pullrequest to be merged or is it unrelated?

**Comment by HoussemNasri** [comment]
@ThiloteE If the fix involves modifying or using one of the following files you should probably wait because they will be deleted, `SidePane`, `GroupSidePane`, `WebSearchPane`, `OpenOfficeSidePanel`, `SidePaneComponent`, there is also `SidePaneType` but it won't be deleted just added more data to it (title, icon, and action).

**Comment by ThiloteE** [comment]
Thank you very much for the fast response.

To avoid creating a false image: I have not written a single code of javascript in my life and am not yet on that level. Maybe one day. By commenting, I intended to provide some sense of coordination and give some loving support to the community. 

This topic is of special interest to me because these error messages seemingly arbitrarily show up (at least arbitrarily for people that don't debug the issue) and I find such bugs very damaging to the reputation of Jabref. The main reasons is that when users look at the user interface for groups, they expect to be able to get a certain functionality based on the way the UI is designed, but these bugs destroy that expectation. Getting a lot of error messages when using groups will probably shy a lot of these users away from Jabref, as long as those users have no clue what causes these error messages.

**Comment by Siedlerchr** [comment]
I just updated an external library which is responsible for the bibtex source and I cannot reproduce it any longer int the latest development build.

**Comment by dsoutw** [comment]
> I just updated an external library which is responsible for the bibtex source and I cannot reproduce it any longer in the latest development build. The new build should be ready in roughly half an hour. https://builds.jabref.org/main/

I confirm that the bug no longer exists in the latest development build.

**Comment by ThiloteE** [comment]
Fix was: #8216
